src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```
@@ -12,11 +15,31 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
+        Transaction tx = null;
+        try (Session session = factory.openSession()) {
+            tx = session.beginTransaction();
+            session.save(actor);
+            tx.commit();
+            return actor;
+        } catch (Exception e) {
+            if (tx != null) {
+                try {
+                    tx.rollback();
+                } catch (Exception ignored) {
+                    throw new DataProcessingException("Can't insert actor " + actor, e);
+                }
+            }
+        }
         return null;
```

The exception handling in this block is incorrect. If an exception occurs in the `try` block, the method should roll back the transaction and then throw a `DataProcessingException`. Your current implementation incorrectly returns `null` if the `rollback()` succeeds, which hides the actual error from the caller. The `throw new DataProcessingException(...)` statement should be the last operation in the main `catch` block to ensure an exception is always thrown on failure.